File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.java
class Solution {
    public int solution(int number, int limit, int power) {
        int answer = 0;
        for (int i = 1; i <= number; i++) {
            int c = count(i);
            if (c > limit) {
                answer += power;
            } else {
                answer += c;
            }
        }
        return answer;
    }

    private int count(int n) {
        int count = 0;
        for (int i = 1; i <= n; i++) {
            if (n % i == 0) {
                count++;
            }
        }
        return count;
    }
}

File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.py
def solution(number, limit, power):
    def divisor_count(n):
        count = 0
        for i in range(1, int(n ** 0.5) + 1):
            if n % i == 0:
                count += 2
                if i*i==n:
                    count -=1
        return count
    answer = 0
    for i in range(1, number + 1):
        if divisor_count(i) > limit:
            answer += power
        else:
            answer += divisor_count(i)
    
    return answer